src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1168 @@ static DamageTable D_80A31B1C = {
     /* Powder Keg     */ DMG_ENTRY(1, 0x0),
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit D_80A31B3C = { 1, 22, 35, 60 };
+static CollisionCheckInfoInit sColChkInfoInit = { 1, 22, 35, 60 };
+
+static TexturePtr sEyeTextures[] = {
+    gChuchuEyeOpenTex,
+    gChuchuEyeHalfTex,
+    gChuchuEyeClosedTex,
+    gChuchuEyeHalfTex,
+};
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A31B54[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(gravity, -2, ICHAIN_CONTINUE),
     ICHAIN_F32(targetArrowOffset, 6000, ICHAIN_STOP),
 };
 
-#endif
+static s32 sVirtualAddrInit = false;
+
+static Color_RGBA8 sBubblePrimColor = { 255, 255, 255, 255 };
+static Color_RGBA8 sBubbleEnvColor = { 150, 150, 150, 0 };
+static Vec3f sBubbleAccel = { 0.0, -0.8, 0.0 };
+
+static Color_RGBA8 sPrimColors[] = {
+    { 255, 255, 255, 255 },
+    { 255, 255, 0, 255 },
+    { 255, 255, 200, 255 },
+    { 225, 200, 255, 255 },
+};
+
+static Color_RGBA8 sEnvColors[] = {
+    { 140, 255, 195, 255 },
+    { 50, 255, 0, 255 },
+    { 255, 180, 0, 255 },
+    { 255, 50, 155, 255 },
+};
+
+static Vec3f sLimbPosMultipliers[] = {
+    { 2000.0, 2000.0, 0.0 }, { -1500.0, 2500.0, -500.0 }, { -500.0, 1000.0, 2500.0 },
+    { 0.0, 4000.0, 0.0 },    { 0.0, 2000.0, -2000.0 },
+};
+
+AnimatedMaterial* sSlimeAnimTex;
+
+/**
+ * Initialize EnSlime instance.
+ */
+void EnSlime_Init(Actor* thisx, PlayState* play) {
+    s32 respawnTimeSeconds;
+    s32 i;
+    EnSlime* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 38.0f);
+
+    this->iceBlockTimer = 255; // 255 is the value used when the ice block is not in use
+    this->eyeTexIndex = EN_SLIME_EYETEX_OPEN;
+    this->wobbleRot.y = 0.0f;
+
+    // Sets maximum distance chuchu will move from its home before turning around.
+    if (this->actor.shape.rot.x <= 0) {
+        this->distLimit = 30000.0f;
+    } else {
+        this->distLimit = (f32)this->actor.shape.rot.x * 40.0f;
```

Comment:
Is this cast required?
```suggestion
        this->distLimit = this->actor.shape.rot.x * 40.0f;
```